src/net/connection.cpp

```diff
@@ -47,8 +55,15 @@ void Connection::handle_write_() {
 }
 
 void Connection::handle_event_(short what) {
+    logger->debug("connection: got bufferevent event: %d", what);
 
     if (what & BEV_EVENT_EOF) {
+        auto input = bufferevent_get_input(bev);
+        if (evbuffer_get_length(input) > 0) {
+            logger->debug("Suppress EOF with data lingering in input buffer");
+            suppressed_eof = true;
+            return;
+        }
```

Comment:
Another error that occurred with fsrn.org: EOF was delivered before "data" with SSL. I fixed this by suppressing EOF when there is buffered data and delivering it after read.

@hellais should I further narrow the check to make sure that we're dealing with a SSL bufferevent by checking whether there is an underlying (i.e. a socket) bufferevent?